### ISSUE TITLE
AtomGroup instant selectors should raise AttributeError and not SelectionError

### ISSUE THREAD

**Issue by GoogleCodeExporter** [MDAnalysis/mdanalysis#220]
What steps will reproduce the problem?
```python
import MDAnalysis as MDA; from MDAnalysis.tests.datafiles import PSF,DCD
u = MDA.Universe(PSF,DCD)
u.atoms.potential_attribute_name
```

What is the expected output? What do you see instead?

It would be less confusing if this raised an *AttributeError* because I suspect  fewer people are using the instant selectors for atom names than are confused when they just mistyped or wrongly guessed an attribute or method name.

I get 

    SelectionError: No atoms or attributes with name potential_attribute_name

I think we should change that error to be in more line with what people expect of python objects.

Original issue reported on code.google.com by `orbeckst` on 19 Feb 2015 at 8:47

**Comment by orbeckst** [comment]
AtomGroup instant selectors now raise AttributeError (Issue #220)
    
- `AtomGroup` instant selectors now raise `AttributeError` and not `SelectionError`
- Note: using string indexing like `AtomGroup['C']` will still raise `SelectionError`      if there is no atom 'C' in the group). `AtomGroup.C` will now raise `AttributeError`.      We could make the former raise `KeyError` instead of `SelectionError` but that seems      less than a problem than the common case in interactive use where mistyping an      attribute or method name gave the cryptic `SelectionError`
- The current implementation just uses another level of indirection and introduces      the method `AtomGroup._get_named_atom(name)`.
- unit test included (check that non-existent attribute raises `AttributeError`)
